context.go

```diff
@@ -313,9 +313,20 @@ func checkRequiredFlags(flags []Flag, context *Context) requiredFlagsErr {
 	var missingFlags []string
 	for _, f := range flags {
 		if rf, ok := f.(RequiredFlag); ok && rf.IsRequired() {
-			key := strings.Split(f.GetName(), ",")[0]
-			if !context.IsSet(key) {
-				missingFlags = append(missingFlags, key)
+			var flagPresent bool
+			var flagName string
+			for _, key := range strings.Split(f.GetName(), ",") {
+				if len(key) > 1 {
+					flagName = key
+				}
+
+				if context.IsSet(strings.TrimSpace(key)) {
```

Comment:
Is it ok if key is a single character?

Reply:
Is there any purpose to check N?

Reply:
That was the bug. For a flag defined as "names, N" we would only check "names" which would not be found as we are passing "N" as the argument.

The fix, makes the library check all the alternate names, in this case "names", "N" to be present in the flagSet before coming to a conclusion that the flag is not present.